### DISCUSSION TITLE
Parse RFC3339 formatted string to core:time

### DISCUSSION THREAD

**Discussion by greenm01** [odin-lang/Odin#2995] [Proposals]
Please consider adding a string parser to core:time, similar to Go's string parser that accepts a RFC3339 formatted string:

https://pkg.go.dev/time#Parse

**Comment by Kelimion** [comment]
Moved to proposals.

**Comment by flysand7** [comment]
![image](https://github.com/odin-lang/Odin/assets/27109464/5daf20df-ae1b-4a29-8326-57ac6c08759f)

The JSON objects produced by JavaScript have dates encoded in RFC3339 format. My guess is that writing/reading date strings of this format might be useful for serialization purposes.

Speaking of which, I don't think we have any way of serializing/deserializing dates in this manner, maybe functions that can understand this format might be useful.

**Comment by Kelimion** [comment]
It's a fair ask. My main question is whether it should return `date: time.Time, tz: string, ok: bool` or just `date: time.Time, ok: bool`, given that we don't support timezones, and they're a massive hassle to support (with Olson's database, etc.)

**Reply by laytan** [reply]
I would say convert to UTC, time.now is also UTC afaik and that would allow you to work on the base assumption of it being UTC.
If/when we add timezones support you can have a struct that uses time.Time as its base and base of that too.

**Reply by flysand7** [reply]
But then the info about timezone is kind of important. The information about the local time of the source of that string may be valuable in some contexts, the RFC gives an example of emails where you might care about whether you can expect an immediate response from the recipient. I think it would be better if it returned timezone, but not as a string as @Kelimion suggested, but rather as a timezone offset, a pair of ints .

I'm not sure why you would suggest strings, but we don't need special handling for timezones aside from having a function that offsets a time by given amount of hours + minutes.

![image](https://github.com/odin-lang/Odin/assets/27109464/42c374b7-f59f-4cd6-aa22-ee504e21aa16)

**Reply by Kelimion** [reply]
I see, you're right. Should then return it as a `time.Time` and `utc_offset_minutes`.

**Comment by Kelimion** [comment]
This is implemented in PR #3292.
